.github/workflows/workflow.yml

```diff
@@ -0,0 +1,71 @@
+name: Python API CI
+
+on: [push, pull_request]
+
+jobs:
+  build-from-source:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install From Source
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install ./PythonAPI
+      - name: Run Simple Test Cases
+        run: |
+          python tests/test_cases.py
+
+  build-from-source-prerelease:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install From Source
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --pre ./PythonAPI
+      - name: Run Simple Test Cases
+        run: |
+          python tests/test_cases.py
+
+  build-from-sdist:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install From Distribution
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install build
+          python -m build --sdist ./PythonAPI
+          python -m pip install --progress-bar off ./PythonAPI/dist/*.tar.gz
+      - name: Run Simple Test Cases
+        run: |
+          python tests/test_cases.py
+      - name: Upload wheel
+        uses: actions/upload-artifact@v3
+        with:
+          name: pycocotools-${{ matrix.python-version }}-${{matrix.os}}-wheel
+          path: ./PythonAPI/dist/*.tar.gz
```

Comment:
This step is titled "Upload wheel", but it actually uploads the sdist, not the wheel.